resources/js/tableofcontent.js

```diff
@@ -1,3 +1,75 @@
+/**
+ * Construct recurseData.str as a JSON Array that contains their header text, 
+ * link, and children headers.
+ * 
+ * References:
+ * https://stackoverflow.com/questions/187619/is-there-a-javascript-solution-to-generating-a-table-of-contents-for-a-page
+ * @param elem DOM element
+ * @param recurseData Object with fields: { level : int, toc : str, count : int, Set : levelSet }
+ */
+function recurseChild(elem, recurseData)
+{
+    if (elem !== "undefined") 
+    {
+        if(elem.nodeName.match(/^H\d+$/) && elem.textContent)
+        {
+            var headerText = elem.textContent.trim();
+            var prevLevel = recurseData.level;
+            var level = elem.nodeName.substr(1);
+            var anchor = "kiwix-toc-" + recurseData.count;
+            var anchorLink = window.location.href.replace(location.hash,"") + '#' + anchor;
```

Comment:
I think that it doesn't make sense to include the full URL with every link. The URL of the page can be provided only once at the root level of the returned JSON object while each TOC entry will include only the anchor.

Reply:
Done